apis/core/v1beta1/function_types.go

```diff
@@ -294,6 +294,7 @@ type FunctionStatus struct {
 	// Addresses holds the addresses that used to access the Function.
 	// +optional
 	Addresses []FunctionAddress `json:"addresses,omitempty"`
+	Output    *Output           `json:"output,omitempty"`
```

Comment:
It's ok to call it Output in Builder status, but we can rename Output to BuilderOutput to be more clear for function status.

Reply:
Move the `output` to `status.builder`?

```
status:
  build:
    resourceHash: "11467684697978232990"
    resourceRef: builder-8zcl4
    state: Succeeded
    output:
      digest: sha256:0ff9c9013ae07a4aa18b036a05ac7f9b0e2e32ff792b1207b10f9abd3108dec3
 

```

Reply:
Adding a builder field to function status might be ok, but do we need all builder status fields?
resourceHash, resourceRef is not necessary. 
There is a build condition filed already, do we need to add the state here too? https://github.com/OpenFunction/OpenFunction/blob/main/apis/core/v1beta1/function_types.go#L292

Reply:
I  mean add the `output` filed to the build condition. The whole status like this.

``
status:
  addresses:
  - type: External
    value: http://function-sample.default.172.31.73.164.sslip.io/
  - type: Internal
    value: http://function-sample.default.svc.cluster.local/
  build:
    resourceHash: "11467684697978232990"
    resourceRef: builder-8zcl4
    state: Succeeded
    output:
      digest: sha256:0ff9c9013ae07a4aa18b036a05ac7f9b0e2e32ff792b1207b10f9abd3108dec3
  route:
    conditions:
    - lastTransitionTime: "2023-01-05T08:32:40Z"
      message: Valid HTTPRoute
      observedGeneration: 6
      reason: Valid
      status: "True"
      type: Accepted
    hosts:
    - function-sample.default.172.31.73.164.sslip.io
    - function-sample.default.svc.cluster.local
    paths:
    - type: PathPrefix
      value: /
  serving:
    lastSuccessfulResourceRef: serving-57k7n
    resourceHash: "12680624694904287116"
    resourceRef: serving-57k7n
    service: serving-57k7n-ksvc-c5l94
    state: Running

``

Reply:
That makes sense

Reply:
If the function has no build and has an image trigger, the trigger controller will update the image's digest. In this case, is it appropriate to put the `output` under `status.build`?

Reply:
or we can use revision instead:
```yaml
status:
    revision:
       imageDigest: xxx
```

We can call the component RevisionWatcher instead of trigger maybe?